test/test_datasets.py

```diff
@@ -44,12 +44,17 @@ def test_downloadurls(self):
 
     def test_openfire(self):
         num_samples = 200
+        img_folder = tempfile.TemporaryDirectory().name # N.B.: different from root
 
         with Path(tempfile.TemporaryDirectory().name) as root:
```

Comment:
Perhaps this
```
with Path(tempfile.TemporaryDirectory().name) as root, Path(tempfile.TemporaryDirectory().name) as img_folder:
    ....
``` 
would be cleaner?

Reply:
Actually I just checked, and we can do:
```
with tempfile.TemporaryDirectory() as root, tempfile.TemporaryDirectory() as img_folder:
    ...
```
and only change the `extract.json` line with  `Path(root).joinpath` rather than `root.joinpath`